tests/components/fan/test_dyson.py

```diff
@@ -91,6 +93,29 @@ def _add_device(devices):
         self.hass.data[dyson.DYSON_DEVICES] = [device_fan, device_non_fan]
         dyson.setup_platform(self.hass, None, _add_device)
 
+    def test_get_state_attributes(self):
+        """Test async added to hass."""
+        device = _get_device_on()
+        component = dyson.DysonPureCoolLinkDevice(self.hass, device)
+        attributes = component.state_attributes
```

Comment:
It would be better to set up the platform, add an entity and get the state attributes for the entity from the core state machine.